### PR TITLE
Replace "gc" with "rebuild" in man pages.

### PR DESCRIPTION
**Pull request by sebastiancarlos**
Also, explain IDs in the context of the working set of tasks.

Related to: https://github.com/GothenburgBitFactory/tw.org/issues/877

### CODE REVIEW

File: doc/man/task-sync.5.in [hunk 1]
```diff
@@ -30,7 +30,7 @@ the existing replica, and run `task sync`.
 
 .SS When to Synchronize
 
-Taskwarrior can perform a sync operation at every garbage collection (gc) run.
+Taskwarrior can perform a sync operation at every rebuild operation.
 This is the default, and is appropriate for local synchronization.
```
Comment (djmitche) [inline doc/man/task-sync.5.in]: Hm, this isn't actually true! I don't remember ever implementing this, so I don't know why I wrote it.
https://github.com/GothenburgBitFactory/taskwarrior/issues/3534 (so, no change required in this PR)

File: doc/man/task.1.in [hunk 2]
```diff
@@ -744,11 +744,12 @@ the online documentation at:
...
 .TP
 .B ID
-Tasks can be specified uniquely by IDs, which are simply the indexes of the
-tasks in the database.  The ID of a task may therefore change, but only when
-a command is run that displays IDs.  When modifying tasks, it is safe to
-rely on the last displayed ID.  Always run a report to check you have the right
-ID for a task. IDs can be given to task as a sequence, for example:
+Tasks can be specified uniquely by IDs, which are the indexes of the "working
+set" of tasks (mostly pending and recurrent tasks).  The ID of a task may
+therefore change, but only when a command is run that displays IDs.  When
```
Comment (djmitche) [inline doc/man/task.1.in]: Specifically, it's reports that will renumber the working set. For example, `task _id /foo/` will show the ID of matching tasks, but not renumber.

File: doc/man/task-sync.5.in (hunk 1 above)
Comment (sebastiancarlos) [reply doc/man/task-sync.5.in]: Got it. This change was removed from the PR.

File: doc/man/task.1.in (hunk 2 above)
Comment (sebastiancarlos) [reply doc/man/task.1.in]: Nice. Updated.

**Review by djmitche — APPROVED** [review]
Great - thank you!

**Pull request merged**